File: Sample Projects/VWAdSuite/VWMasterApplication/VWMasterApplication/VWVerveServerViewController.m
//
//  VWVerveServerViewController.m
//  VWMasterApplication
//
//  Created by Christina Long on 9/25/15.
//  Copyright (c) 2015 Christina Long. All rights reserved.
//

#import "VWVerveServerViewController.h"
#import <VWAdvertView.h>
#import <VWInterstitialAd.h>

@interface VWVerveServerViewController () <VWInterstitialAdDelegate, VWAdvertViewDelegate>

@property (strong, nonatomic) VWAdvertView *bannerAdView;
@property (strong, nonatomic) VWAdvertView *inlineAdView;
@property (strong, nonatomic) VWInterstitialAd *interstitial;

@end


#pragma mark - View Controller Lifecycle

@implementation VWVerveServerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self addBannerAdView];
    [self addInlineAdView];
}

#pragma mark - Banner Ad Code

-(void)addBannerAdView {
    //determine size for device
    VWAdSize size = (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) ? kVWAdSizeLeaderboard : kVWAdSizeBanner;
    
    self.bannerAdView = [[VWAdvertView alloc] initWithSize:size];
    self.bannerAdView.adPosition = VWAdvertPositionBottom;
    self.bannerAdView.delegate = self;
    self.bannerAdView.backgroundColor = [UIColor grayColor];

    //determine size/frame for banner Ad
    CGRect bounds = self.view.bounds;
    CGRect adFrame = CGRectZero;
    
    adFrame.size = [self.bannerAdView sizeThatFits:bounds.size];
    adFrame.origin.y = bounds.size.height - adFrame.size.height - self.tabBarController.tabBar.frame.size.height;
    
    self.bannerAdView.frame = adFrame;
    
    //ad the banner ad to the view
    [self.view addSubview:self.bannerAdView];
}

- (IBAction)requestBannerAd:(id)sender {
    [self.bannerAdView loadRequest:[VWAdRequest requestWithContentCategoryID:VWContentCategoryNewsAndInformation]];
}


#pragma mark - Inline Ad Code

-(void)addInlineAdView {
    //add the inline ad view
    self.inlineAdView = [[VWAdvertView alloc] initWithSize:kVWAdSizeMediumRectangle];
    self.inlineAdView.adPosition = VWAdvertPositionInline;
    self.inlineAdView.delegate = self;
    self.inlineAdView.backgroundColor = [UIColor grayColor];

    //determine frame and size for the view
    CGRect bounds = self.view.bounds;
    CGRect adFrame = CGRectZero;
    
    adFrame.size = [self.inlineAdView sizeThatFits:bounds.size];
    self.inlineAdView.frame = CGRectMake((bounds.size.width - adFrame.size.width)/2, (bounds.size.height - adFrame.size.height)/2, adFrame.size.width, adFrame.size.height);
    
    [self.view addSubview:self.inlineAdView];
}

- (IBAction)requestInlineAd:(id)sender {
    [self.inlineAdView loadRequest:[VWAdRequest requestWithContentCategoryID:VWContentCategoryNewsAndInformation]];
}

#pragma mark - VWAdvertViewDelegate

- (void)advertViewDidReceiveAd:(VWAdvertView *)adView {
    //handle successful ad
}

- (void)advertView:(VWAdvertView *)adView didFailToReceiveAdWithError:(NSError *)error {
    //handle error condition
}

#pragma mark - Interstitial Ad Code

- (IBAction)requestIntersititalAd:(id)sender {
    VWInterstitialAdSize interstitialAdSize = (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) ? VWInterstitialAdSizePad : VWInterstitialAdSizePhone;
    
    self.interstitial = [[VWInterstitialAd alloc] initWithSize:interstitialAdSize];
    self.interstitial.delegate = self;
    
    VWAdRequest *adRequest = [VWAdRequest requestWithContentCategoryID:VWContentCategoryNewsAndInformation];
    
    [self.interstitial loadRequest:adRequest];
}

#pragma mark - VWInterstitialAdDelegate

- (void)interstitialAdReceiveAd:(nonnull VWInterstitialAd *)interstitialAd {
    [self.interstitial presentFromViewController:self];
}

- (void)interstitialAd:(nonnull VWInterstitialAd *)interstitialAd didFailToReceiveAdWithError:(nullable NSError *)error {
    self.interstitial = nil;
}

- (void)interstitialAdWillPresentAd:(nonnull VWInterstitialAd *)interstitialAd {
}

- (void)interstitialAdDidDismissAd:(nonnull VWInterstitialAd *)interstitialAd {
    self.interstitial = nil;
}

- (void)interstitialAdWillLeaveApplication:(nonnull VWInterstitialAd *)interstitialAd {
    self.interstitial = nil;
}

@end

File: Sample Projects/VWAdSwiftExample/VWAdSwiftExample/VWVerveServerViewController.swift
//
//  VWVerveServerViewController.swift
//  VWAdSwiftExample
//
//  Created by Srdan Rasic on 25/09/14.
//  Copyright (c) 2014 Verve Wireless. All rights reserved.
//

import UIKit
import VerveAd

class VWVerveServerViewController: UIViewController, VWAdvertViewDelegate, VWInterstitialAdDelegate {
  
  var bannerAdView : VWAdvertView?
  var inlineAdView : VWAdvertView?
  var interstitial: VWInterstitialAd?
  
  //MARK: View Lifecycle
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    addBannerAdView()
    addInlineAdView()
  }
  
  //MARK: - Banner Ads
  
  func addBannerAdView() {
    guard self.bannerAdView == nil else { return }
    
    let adSize = UI_USER_INTERFACE_IDIOM() == .pad ? kVWAdSizeLeaderboard : kVWAdSizeBanner
    
    guard let tabBarHeight = tabBarController?.tabBar.frame.size.height else {
      return
    }
    
    let bannerAdView = VWAdvertView(size: adSize)
    bannerAdView.adPosition = VWAdvertPositionBottom
    bannerAdView.delegate = self
    bannerAdView.backgroundColor = .gray
    
    let bounds = view.bounds
    var adFrame = CGRect.zero
    
    adFrame.size = bannerAdView.sizeThatFits(bounds.size)
    adFrame.origin.y = bounds.size.height - adFrame.size.height - tabBarHeight
    
    bannerAdView.frame = adFrame
    
    view.addSubview(bannerAdView)
    
    self.bannerAdView = bannerAdView
  }
  
  @IBAction func requestBannerAd() {
    bannerAdView?.load(VWAdRequest(contentCategoryID : .newsAndInformation))
  }
  
  
  //MARK: - Inline Ads
  
  func addInlineAdView() {
    guard self.inlineAdView == nil else { return }
    
    let inlineAdView = VWAdvertView(size: kVWAdSizeMediumRectangle)
    
    inlineAdView.adPosition = VWAdvertPositionInline
    inlineAdView.delegate = self
    inlineAdView.backgroundColor = .gray
    
    let bounds = view.bounds
    var adFrame = CGRect.zero
    
    adFrame.size = inlineAdView.sizeThatFits(bounds.size)
    inlineAdView.frame = CGRect(
      x: (bounds.size.width - adFrame.size.width)/2,
      y: (bounds.size.height - adFrame.size.height)/2,
      width: adFrame.size.width,
      height: adFrame.size.height)
    
    view.addSubview(inlineAdView)
    
    self.inlineAdView = inlineAdView
  }
  
  @IBAction func requestInlineAd() {
    inlineAdView?.load(VWAdRequest(contentCategoryID : .newsAndInformation))
  }
  
  //MARK: - Interstitial Ads
  
  @IBAction func requestInterstitialAd() {
    let adSize = UI_USER_INTERFACE_IDIOM() == .pad ? VWInterstitialAdSizePad : VWInterstitialAdSizePhone
    
    interstitial = VWInterstitialAd(size: adSize)
    interstitial?.delegate = self
    
    let adRequest = VWAdRequest(contentCategoryID: .newsAndInformation)
    
    interstitial?.load(adRequest)
  }
  
  // MARK: - VWAdvertViewDelegate
  
  func advertViewDidReceiveAd(_ adView: VWAdvertView) {
    NSLog("adViewDidReceiveAd: %@", adView)
  }
  
  func advertView(_ adView: VWAdvertView, didFailToReceiveAdWithError error: Error?) {
    if let error = error as NSError? {
      NSLog("didFailToReceiveAdWithError: %@", error)
    }
  }
  
  
  // MARK: - VWInterstitialDelegate
  func interstitialAdReceive(_ interstitialAd: VWInterstitialAd) {
    NSLog("interstitialAdReceiveAd: %ld", interstitialAd)
    
    interstitial?.present(from: self)
  }
  
  func interstitialAd(_ interstitialAd: VWInterstitialAd, didFailToReceiveAdWithError error: Error?) {
    if let error = error as NSError? {
      NSLog("dismissInterstitialViewControllerVerveInterstitial: %@", error)
    }
    
    interstitial = nil
  }
  
  func interstitialAdWillPresent(_ interstitialAd: VWInterstitialAd) {
    NSLog("interstitialAdWillPresentAd: %ld", interstitialAd);
  }
  
  func interstitialAdDidDismiss(_ interstitialAd: VWInterstitialAd) {
    NSLog("interstitialAdDidDismissAd: %ld", interstitialAd);
    interstitial = nil;
  }
  
  func interstitialAdWillLeaveApplication(_ interstitialAd: VWInterstitialAd) {
    NSLog("interstitialAdWillLeaveApplication: %ld", interstitialAd);
    interstitial = nil;
  }
}

